lib/notification/adapter/telegram.js

```diff
@@ -1,8 +1,49 @@
 import { markdown2Html } from '../../services/markdown.js';
 import { getJob } from '../../services/storage/jobStorage.js';
 import fetch from 'node-fetch';
+import pThrottle from 'p-throttle';
+
 const MAX_ENTITIES_PER_CHUNK = 8;
-const RATE_LIMIT_INTERVAL = 1010;
+const RATE_LIMIT_INTERVAL = 1000;
+const chatThrottleMap = new Map();
+
+function cleanupOldThrottles() {
```

Comment:
The cleanupOldThrottles function deletes entries from chatThrottleMap if now - chatThrottle.lastUsedAt > RATE_LIMIT_INTERVAL. Since RATE_LIMIT_INTERVAL is 1 sec, entries are removed if they haven't been used in the last second. However, I think this is too aggressive for a rate-limiting system, as it could delete a throttle entry while it's still actively limiting requests. For example, if a chat sends a message at t=0ms and another at t=500ms, the throttle entry might be deleted at t=1001ms during a subsequent call, even though the throttle is still needed to enforce the rate limit for ongoing messages.